src/referral_maintenance.rs

```diff
@@ -120,32 +129,51 @@ async fn download_referral_from_ipfs_and_store_in_referral_store(
     match get_cid_from_app_data(hash) {
         Ok(cid) => {
             tracing::debug!("cid for hash {:?} is {:?}", hash, cid);
-            match get_ipfs_file_and_read_referrer(cid.clone()).await {
-                Ok(referrer) => {
-                    tracing::debug!("Adding the referrer {:?} for the hash {:?}", referrer, hash);
+            match get_ipfs_file_and_read_app_data(cid.clone()).await {
+                Ok(app_data) => {
+                    if let Some(referrer) = app_data.read_referrer() {
+                        tracing::debug!(
+                            "Adding the referrer {:?} for the hash {:?}",
```

Comment:
I think its fine if you also print here the complete appData

Reply:
Cool, I will replace this if block with a simple log like ("found corresponding content {}", app_data)